lisp/mastodon-tl.el

```diff
@@ -212,22 +212,72 @@ also render the html"
   (replace-regexp "\n\n\n | " "\n | " nil (point-min) (point-max))
   (mastodon-media--inline-images))
 
-(defun mastodon-tl--more-json (timeline id)
-  "Return JSON for TIMELINE before ID."
-  (let ((url (mastodon-http--api (concat "timelines/"
-                                         timeline
-                                         "?max_id="
-                                         (number-to-string id)))))
+(defun mastodon-tl--get-endpoint (buffer-string)
+  "Match BUFFER-STRING against buffer-names.
+
+Returns the matching plist"
+  (let* ((mastodon-buffer-endpoints ;; can be made a global var
+          '((buffer-name "home" endpoint "timelines/home"
+                         update-function mastodon-tl--timeline)
+            (buffer-name "local" endpoint "timelines/public?local=true"
+                         update-function mastodon-tl--timeline)
+            (buffer-name "federated" endpoint "timelines/public"
+                         update-function mastodon-tl--timeline)
+            (buffer-name "public?local=true"
+                         endpoint "timelines/public?local=true"
+                         update-function mastodon-tl--timeline)
+            (buffer-name "public"
+                         endpoint "timelines/public"
+                         update-function mastodon-tl--timeline)
+            (buffer-name "tag/.*"
+                         endpoint (lambda (x) (concat "timelines/" x))
+                         update-function mastodon-tl--timeline
+                         endpoint-function)
+            (buffer-name "notifications"
+                         endpoint "notifications"
+                         update-function
+                         mastodon-notifications--notifications)))
+         (match-endpoint-string
+             (lambda (endpoint-plist)
+               (when (string-match
+                      (plist-get endpoint-plist 'buffer-name) buffer-string)
+                 (unless (functionp
+                          (plist-get endpoint-plist 'update-function))
+                   (error
+                    (format "%s not a function"
+                            (plist-get endpoint-plist 'update-function))))
+                 (cond
+                  ((stringp (plist-get endpoint-plist 'endpoint))
+                   endpoint-plist)
+                  ((functionp (plist-get endpoint-plist 'endpoint))
+                   (plist-put end 'endpoint
```

Comment:
This still says `end` instead of `endpoint-plist`.

This strongly suggests that it is not covered by any of your tests.   ;-)